discussment-core/src/main/java/org/danekja/discussment/core/service/imp/DefaultPostService.java

```diff
@@ -107,6 +109,65 @@ public IDiscussionUser getPostAuthor(Post post) throws DiscussionUserNotFoundExc
         }
     }
 
+    public Post getLastPost(Discussion discussion) throws AccessDeniedException{
+        if(accessControlService.canViewPosts(discussion)) {
+            Post lastPost = null;
+
+            for (Post post : postDao.getPostsByDiscussion(discussion)) {
+                if (lastPost == null) {
+                    lastPost = post;
+                }
+
+                Post a = lastPost.getLastPost();
+                Post b = post.getLastPost();
+
+                if (b.getCreated().compareTo(a.getCreated()) > 0) {
+                    lastPost = b;
+                } else {
+                    lastPost = a;
+                }
+            }
+            return lastPost;
+        } else {
+            throw new AccessDeniedException(Action.VIEW, getCurrentUserId(),discussion.getId(), PermissionType.POST);
+        }
+    }
+
+    public List<Post> getReplies(Post post) throws AccessDeniedException{
+        if(accessControlService.canViewPost(post)){
+            return postDao.getRepliesForPost(post);
+        } else {
+            throw new AccessDeniedException(Action.VIEW, getCurrentUserId(),post.getId(), PermissionType.POST);
+        }
+    }
+
+    public int getNumberOfPosts(Discussion discussion) throws AccessDeniedException{
+        if(accessControlService.canViewPosts(discussion)) {
+            int numberOfPosts = 0;
+            for (Post post : postDao.getPostsByDiscussion(discussion)) {
```

Comment:
This is terribly inefficient. Add counting method to the postDao with query returning number of posts belonging to the given discussion.

Remove unussed getNumberOfReplies methods